final_task/rss_reader/args_parser.py

```diff
@@ -0,0 +1,95 @@
+"""
+This module process the input arguments
+"""
+
+import argparse
+import re
+import logs
+import sys
+import datetime
+import os
+
+
+def get_parse(args_in='') -> dict:
+    """This function takes the arguments from command line"""
+
+    parser = argparse.ArgumentParser(prog='RSSTaker', description='RSS reader. Takes the arguments from command line.')
+
+    parser.add_argument('url', type=str, nargs='?', default='url', help='Link to RSS channel(line without spaces).')
+    parser.add_argument('-v', '--version', action='version', version='%(prog)s 3.0', help='Print version info.')
+    parser.add_argument('-j', '--json', action='store_const', const=True, help='Print result as json in stdout.')
+    parser.add_argument('-b', '--verbose', action='store_const', const=True, help='Print all logs in stdout.')
+    parser.add_argument('-l', '--limit', type=int, help='Limit of news topics (natural number).')
+    parser.add_argument('-d', '--date', type=int, help='Date to print news from history, yyyymmdd (natural number).')
+    parser.add_argument('-p', '--pdf', type=str, help=r'Convert news to pdf (use -p [path where to store\].')
+    parser.add_argument('-hl', '--html', type=str, help=r'Convert news to html (use -hl [path where to store\].')
+
+    if args_in:
+        try:
+            args = parser.parse_args(args_in)
+            logs.log_init_args(args)
+        except SystemExit:
+            logs.log_err_init_args(args_in)
+            logs.log_err_exit()
+            parser.exit()
+    else:
+        try:
+            args = parser.parse_args()
+            logs.log_init_args(args)
+        except SystemExit:
+            logs.log_err_init_args(sys.argv)
+            logs.log_err_exit()
+            parser.exit()
+
+    return {'url': args.url,
+            'json': args.json,
+            'verbose': args.verbose,
+            'limit': args.limit,
+            'date': args.date,
+            'pdf': args.pdf,
+            'html': args.html}
```

Comment:
A question:

You already have all arguments inside `args` object. Do you really need to make this dict here? Why not just pass `args` everywhere instead?

Reply:
I use this data in the code in rss_reader.py module. It is more convenient for me to use it in this form. In my opinion it make my code easy to read.